plugins/nlp/nlp.cpp

```diff
@@ -57,12 +58,15 @@ namespace nlp {
 
     ECL_NLP_API void ECL_NLP_CALL AnalyzeText(size32_t & tgtLen, char * & tgt, size32_t anaLen, const char * ana, size32_t txtLen, const char * txt)
     {
-        CriticalBlock block(cs);
-        if (nlpEng == NULL) {
-            nlpEng = new NLPEng();
+        {
+            CriticalBlock block(cs);
+            if (nlpEng == NULL) {
+                nlpEng = new NLPEng();
+            }
+            activeEngine = nlpEng;
         }
         ostringstream sso;
-        tgtLen = nlpEng->nlpEngAnalyze(ana,txt,sso);
+        tgtLen = activeEngine->nlpEngAnalyze(ana,txt,sso);
```

Comment:
sorry, my code was a bit confusing - this could be nlpEng, and no need for variable activeEngine.
(It was an extract from a more complex solution to the same problem.)